### DISCUSSION TITLE
Can we generate classroom pictures with SD?

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#3270] [Q&A] [answered]
The big idea is that we have teachers, students, and administrators take structured surveys to determine key design patterns they should incorporate into their classroom; we have 21 design patterns right now, but will likely add more in the future. Coming out of the survey, we stack rank the 21 design patterns and show the top 8 to the teacher.
We also have the teacher take a 'before' picture of their classroom. In doing this project, we would like to show a potential 'after' picture that incorporates some of the top design patterns into the picture. This could be adding specific furniture, raising the ceiling, etc.
My current thought is that we may need to train the model on elements of the 21 design patterns, so we can incorporate this into the prompt & input image. My gut says this would be a LORA model, but I am wanting to get your thoughts on the best approach.
I have done enough work on Stable Diffusion & training to know that I am not an expert and need to rely on people with **more expertise.**

Stable Diffusion Inputs:
- Picture of a classroom (user taken)
- Prompt with standard wording
- Additional information based on a bunch of survey information and design patterns.

Stable Diffusion Output:

- Image from stable diffusion that is the same room as the user-taken picture, but redesigned with the information from the prompt and survey information. It should be a photorealistic style. I will need your help to understand if it's better to get 1 image or multiple options back from SD.
![01a_2024-03-21_Kawartha_Before_ar4_3](https://github.com/user-attachments/assets/1ffe9531-161a-47e1-ba84-55cad7138178)

Greatly appreciate if you experts can guide me how I can build this succ
![Uploading 02a_202403-22 North_Jr_Hi_1_Before_ar3-3.jpg…]()

![Uploading 01b_2024-04-22_Kawartha_after_ar16-9.png…]()
![Uploading 02b_2024-04-22 02_North_Jr_Hi_After_ar16-9.png…]()
![Uploading 01b_2024-04-22_Kawartha_after_ar16-9.png…]()

essfully.
[Design Pattern Details.pdf](https://github.com/user-attachments/files/16197389/Design.Pattern.Details.pdf)

**Comment by mashb1t** [comment, marked as answer]
None of the additional were actually fully uploaded, you have to be more patient next time.
I'd propose to go for
- 1 LoRA trained for each pattern or 1 lora in total trained with all patterns but using different keywords for each bucket, then use it with inpainting or generating in general. Keep in mind that the rule of thumb is to train each concept (pattern) with at least 20 images to have sufficient training data for the AI to work with.
- only inpainting with good prompts

**Reply by oldhand7** [reply]
Thanks. mashb1t. 
[Uploading 21 Design Patterns 2024-07-09 (2).pdf…]()
Could you review this PDF and let me know if it's possible with Foocus?
Thank you in advance.

**Reply by mashb1t** [reply]
As always, your link is broken. You have to wait until fully uploaded!
Also i can't and won't, no time for personal support of this depth.

**Reply by oldhand7** [reply]
![01a_2024-03-21_Kawartha_Before_ar4_3](https://github.com/user-attachments/assets/87d600c4-23a9-4c89-b243-a3e5f0e274d1)
![01b_2024-04-22_Kawartha_after_ar16-9](https://github.com/user-attachments/assets/c645bcc6-9884-4e43-b121-80d10ccbb1d9)
![03a_2024-03-22_GlenLake_ES_Before_ar4-3](https://github.com/user-attachments/assets/c345d993-ff6a-49f9-8d72-944d2c6809f4)
![04a_2024-03-22_Kawartha-2_Before_ar4](https://github.com/user-attachments/assets/073fdd19-f59f-4f0c-b1cf-609213bc098e)
![04b_2024-04-22 01 Kawartha 2 After](https://github.com/user-attachments/assets/46d8307d-139f-4a71-8c74-f26c7a180805)
![05a_2024-03-22_Eisenhower_Before_ar4](https://github.com/user-attachments/assets/be731564-5f78-4fa9-90b8-d8d5aba42b0f)

![02b_2024-04-22 02_North_Jr_Hi_After_ar16-9](https://github.com/user-attachments/assets/6ab2ebc3-ced2-4da9-9a7b-91fb8f1df6e3)

![02a_202403-22 North_Jr_Hi_1_Before_ar3-3](https://github.com/user-attachments/assets/a572cd95-fe58-4fbb-a5fd-4b0ac3d78471)





Those I 've fully uploaded. Thank you for your understanding.

**Comment by IPv6** [comment]
Frankly such questions should be asked on Reddit - there is a lot of subreddits with users trained a lot of stuff by themselves. try r/stablediffusion or r/civitai. Fooocus is about inference, you will hardly find extensive advices on training here
